src/google/protobuf/compiler/cpp/cpp_message.cc

```diff
@@ -1220,7 +1220,12 @@ void MessageGenerator::GenerateClassDefinition(io::Printer* printer) {
       "  return reinterpret_cast<const $classname$*>(\n"
       "             &_$classname$_default_instance_);\n"
       "}\n"
+	  #ifdef __CUDACC__
```

Comment:
Actually this might not work, because this ifdef will only affect protoc, which is just a plain binary unrelated to Cuda. For it to work it would probably have to go in the generated code.